__tests__/tokens_utils.js

```diff
@@ -251,3 +251,229 @@ test('Token withdraw', () => {
+        token_data: 129,
+        script: 'dqkU1YP+t130UoYD+3ys9MYt1zkWeY6IrA==',
+        decoded: {
+          type: 'P2PKH',
+          address: 'Wi8zvxdXHjaUVAoCJf52t3WovTZYcU9aX6',
+          timelock: null,
+        },
+        token: '006b972eb0d560b168f1d544c7d772e27860a9bc2f21ab7a2a992ea5ad3a99d8',
+        spent_by: null,
+        selected_as_input: false,
+      },
+      {
+        value: 2,
+        token_data: 129,
+        script: 'dqkULlcsARvA+pQS8qytBr6Ryjc/SLeIrA==',
+        decoded: {
+          type: 'P2PKH',
+          address: 'WSu4PZVu6cvi3aejtG8w7bomVmg77DtqYt',
+          timelock: null,
+        },
+        token: '006b972eb0d560b168f1d544c7d772e27860a9bc2f21ab7a2a992ea5ad3a99d8',
+        spent_by: null,
+        selected_as_input: false,
+      },
+    ],
+    parents: [
+      '00d11791843e284ce6ad833ee3ee2f3b88cacb6d3888be0eae400abc8d643b8c',
+      '54165cef1fd4cf2240d702b8383c307c822c16ca407f78014bdefa189a7571c2',
+    ],
+    token_name: 'New NFT',
+    token_symbol: 'NNFT',
+    tokens: [],
+  };
+  const cloneNftSample = () => JSON.parse(JSON.stringify(sampleNftTx));
+
+  it('should return false for an empty transaction object', () => {
+    expect.assertions(1);
+    expect(tokens.isNFTToken(null)).toBe(false);
+  });
+
+  it('should return false for a default transaction version', () => {
+    expect.assertions(1);
+    const tx = cloneNftSample();
+    tx.version = DEFAULT_TX_VERSION;
+    expect(tokens.isNFTToken(tx)).toBe(false);
+  });
+
+  it('should return false for a token-creating tx with less than 2 outputs', () => {
+    expect.assertions(1);
+    const tx = cloneNftSample();
+    tx.outputs.length = 1;
+    expect(tokens.isNFTToken(tx)).toBe(false);
+  });
+
+  it('should return false for a fee output with wrong data', () => {
+    expect.assertions(2);
+    const tx = cloneNftSample();
+
+    // Wrong Value
+    tx.outputs[0].value = 2;
+    expect(tokens.isNFTToken(tx)).toBe(false);
+
+    // Wrong Token Data
+    tx.outputs[0].value = 1;
+    tx.outputs[0].token_data = 1;
+    expect(tokens.isNFTToken(tx)).toBe(false);
+  });
+
+  it('should return false for having a non-standard output script', () => {
+    expect.assertions(2);
+    const tx = cloneNftSample();
+
+    // Script too large
+    tx.outputs[1].script = 'a'.repeat(257);
+    expect(tokens.isNFTToken(tx)).toBe(false);
+
+    // Non-standard output type
+    tx.outputs[1].script = sampleNftTx.outputs[1].script;
+    tx.outputs[1].decoded.type = 'data';
+    expect(tokens.isNFTToken(tx)).toBe(false);
+  });
+
+  it('should return false for having a non-standard token on outputs', () => {
+    expect.assertions(1);
+    const tx = cloneNftSample();
+
+    // Changing one of the outputs to be of another token
+    tx.outputs[1].token = '54165cef1fd4cf2240d702b8383c307c822c16ca407f78014bdefa189a7571c2';
+    expect(tokens.isNFTToken(tx)).toBe(false);
+  });
+
+  it('should return false for having a non-standard token on inputs', () => {
```

Comment:
This concept of "standard" was better discussed in [a comment above](https://github.com/HathorNetwork/hathor-wallet-lib/pull/378#discussion_r898556293) and it was removed from the code to avoid ambiguity.